perceval/components/abstract_processor.py

```diff
@@ -47,21 +47,25 @@ def __init__(self):
         self.name: str = ""
         self._parameters: Dict = {}
 
-        self._in_ports: Dict = {}
-        self._out_ports: Dict = {}
         self._thresholded_output: bool = False
         # Mode post selection: expect at least # modes with photons in output
         self._min_mode_post_select = None
-        self._post_select = None
+
+        self._reset_circuit()
+
+    def _reset_circuit(self):
```

Comment:
IMO, the variables which are the "Circuit" should be in a dedicated class

Reply:
Let's discuss this during a specific meeting, as this is too big a request to be implemented in this branch.